src/Creatable.js

```diff
@@ -141,6 +122,51 @@ export const makeCreatableSelect = (SelectComponent: ComponentType<*>) =>
       }
       onChange(newValue, actionMeta);
     };
+    
+    isValidNewOption (inputValue: string, selectValue: OptionsType, selectOptions: OptionsType) {
+    	if (typeof this.props.isValidNewOption === 'function') {
+    	      return this.props.isValidNewOption(inputValue, selectValue, selectOptions);
+    	    } else {
+    	return !(
+    	      !inputValue ||
+    	      selectValue.some(option => this.compareOption(inputValue, option)) ||
+    	      selectOptions.some(option => this.compareOption(inputValue, option))
+    	    )
+    	    }
+    }
+	 
+ compareOption(inputValue, option){
+	 if (typeof this.props.compareOption === 'function') {
```

Comment:
It seems wasteful to check and delegate here, since this will be called on every element.

Instead I would leave `compareOption` (perhaps as `defaultCompareOption`) in the outer scope and put the check in isValidNewOption.